state/remoteapplication.go

```diff
@@ -360,9 +368,10 @@ func (s *RemoteApplication) removeOps(asserts bson.D) []txn.Op {
 			Remove: true,
 		},
 		removeStatusOp(s.st, s.globalKey()),
-		r.removeRemoteEntityOp(s.SourceModel(), s.Tag()),
 	}
-	return ops
+	tokenOps := r.removeRemoteEntityOps(s.SourceModel(), s.Tag(), token)
```

Comment:
IMO it would be neater and less error-prone (less unintentional sweeping under the rug) if `removeRemoteEntityOps` expected a non-empty token, and we only called it here in the case where GetToken above returned a nil error.

I guess it becomes moot if we move the token into the remote application doc though? Is that coming soon?